storage/src/backend/connection.rs

```diff
@@ -626,11 +626,19 @@ impl Connection {
         } else {
             None
         };
+        // get pool size from envvar
+        let pool_max_idle_per_host = match env::var("REGISTRY_CLIENT_POOL_MAX_IDLE_PER_HOST") {
+            Ok(val) => val.parse::<usize>().unwrap_or(20),
+            Err(_) => 20,
+        };
 
         let mut cb = Client::builder()
             .timeout(timeout)
             .connect_timeout(connect_timeout)
-            .redirect(Policy::none());
+            .redirect(Policy::none())
+            .use_rustls_tls()
```

Comment:
why `use_rustls_tls()` here? It introduces heavy dependencies.

Reply:
How about using `native-tls` instead? We already have vendored openssl-sys crate.
```
openssl = { version = "0.10.55", features = ["vendored"] }
```

Reply:
it's required for http/2, as far as I know native-tls does not provide for the ALPN required for http/2

Reply:
would this be acceptable?

Reply:
It should be acceptable I think.

Reply:
Should I make any other changes to this then?

Reply:
How about making the env `REGISTRY_CLIENT_POOL_MAX_IDLE_PER_HOST` value as the one option of ConnectionConfig?